File: fortran/mandelbrot.f90
program mandelbrot
  implicit none
  integer, parameter :: width = 16000
  integer, parameter :: height = 6000
  integer, parameter :: iterations = 64
  complex, parameter :: min = (-2.1, -1.2)
  complex, parameter :: max = (0.6, 1.2)

  complex, parameter :: inc_real = cmplx( real(max - min)/width, 0.0 )
  complex, parameter :: inc_imag = cmplx( 0.0, aimag(max - min)/height )

  character (len=18) :: colors
  complex :: row_val
  integer :: row
  integer :: col
  complex :: z
  complex :: c
  integer :: color_idx
  integer :: iter_count

  real abs2
  abs2( z ) = real(z)**2 + aimag(z)**2

  colors  = 'abcdefhij.-+*%#$@ '
  do row = 0, height
     row_val = row * inc_imag + min
     do col = 0, width
        c = col * inc_real + row_val
        z = 0
        inner_loop: do iter_count = 0, iterations
           z = z * z + c
           if (abs2(z) > 4.0) then
              exit inner_loop
           end if
        end do inner_loop
        color_idx = 1 + ((len(colors)-1) * iter_count) / iterations
        write(*, fmt="(A)", advance="no") colors(color_idx:color_idx)
     end do
     print *, ''
  end do
end program mandelbrot

File: pascal/mandelbrot.pas
program mandelbrot;
uses uComplex, math;
const
   width           = 16000;
   height          = 6000;
   min  : complex  = (re : -2.1; im : -1.2);
   max  : complex  = (re : 0.6; im : 1.2);
   iterations      = 64;
   colors : string = 'abcdefhij.-+*%#$@ ';

var
   inc_real   : complex;
   inc_imag   : complex;
   row        : integer;
   col        : integer;
   row_val    : complex;
   z          : complex;
   c          : complex;
   color_idx  : integer;
   iter_count : integer;

   function cnorm(z : complex) : real;
   begin
      cnorm := z.re * z.re + z.im * z.im;
   end;


   begin
      inc_real := cinit((max - min).re / width,  0.0);
      inc_imag := cinit( 0.0 ,  (max-min).im / height);

      for row := 0 to height do
      begin
         row_val := min + row * inc_imag;
         for col := 0 to width do
         begin
            c:= row_val + col * inc_real;
            z := c;
            iter_count := 0;
            repeat
               iter_count := iter_count + 1;
               z := z*z + c;
            until ((cnorm(z)> 4.0) or (iter_count > iterations));
            color_idx := 1 + floor(((Length(colors)-1) * iter_count) / iterations);
            write(colors[color_idx]);
         end;
         WriteLn('');
      end;
   end.
